### PR TITLE
Jetpack Sync: Add sync action for theme edit

### PR DESCRIPTION
**Pull request by gititon**
Jetpack Sync: Add sync action for theme edit

#### Changes proposed in this Pull Request:

Adds a sync action for theme edit

#### Testing instructions:

phpunit runs a new test that tests the functionality

### CODE REVIEW

File: sync/class.jetpack-sync-module-themes.php [hunk 1]
```diff
@@ -27,9 +27,13 @@ public function init_listeners( $callable ) {
 
 	public function detect_theme_edit( $redirect_url ) {
 		$url = wp_parse_url( $redirect_url );
-		if ( '/wp-admin/theme-editor.php' !== $url['path'] ) {
+		$expected_url = admin_url() . 'theme-editor.php';
```
Comment (enejb) [inline sync/class.jetpack-sync-module-themes.php]: `admin_url( 'theme-editor.php' );` should work as well.

File: tests/php/sync/test_class.jetpack-sync-themes.php [hunk 2]
```diff
@@ -123,45 +123,56 @@ public function test_changed_theme_is_synced() {
...
-		delete_theme( $theme_stylesheet );
+		//Test Edit Theme
 
+		/**
+		 * This filter is already documented in wp-includes/pluggable.php
+		 *
+		 * @since 1.5.1
+		 */
+		$_POST['newcontent'] = 'foo';
+		apply_filters( 'wp_redirect', 'theme-editor.php?file=style.css&theme=' . $theme_slug . '&scrollto=0&updated=true');
```
Comment (enejb) [inline tests/php/sync/test_class.jetpack-sync-themes.php]: '&scrollto=0&updated=true' ); missing space.

Comment (enejb) [inline tests/php/sync/test_class.jetpack-sync-themes.php]: Should we try testing this with different types of urls passed into the redirect? 
Such as the full url?

File: tests/php/sync/test_class.jetpack-sync-themes.php [hunk 3]
```diff
@@ -263,4 +274,23 @@ public function test_widgets_changes_get_synced() {
...
+			array(
+				'slug'   => $slug,
+			)
+		);
+
+		if ( is_wp_error( $api ) ) {
+			wp_die( $api );
+		}
+
+		$upgrader = new Theme_Upgrader( new Test_Upgrader_Skin( compact('title', 'nonce', 'url', 'theme') ) );
```
Comment (enejb) [inline tests/php/sync/test_class.jetpack-sync-themes.php]: There are some spaces missing here as well.

File: sync/class.jetpack-sync-module-themes.php [hunk 4]
```diff
@@ -23,6 +25,44 @@ public function init_listeners( $callable ) {
 		add_action( 'jetpack_widget_reordered', $callable );
 	}
 
+	public function detect_theme_edit( $redirect_url ) {
+		$url = wp_parse_url( admin_url( $redirect_url ) );
```
Comment (enejb) [inline sync/class.jetpack-sync-module-themes.php]: $redirect_url is a full url. such as google.com 
we would expected $url = wp_parse_url( "http://example.com/wp-admin/http://google.com" );
is that what we want?

Comment (gititon) [inline sync/class.jetpack-sync-module-themes.php]: $redirect_url is just the file name when/where we need to examine it, see https://github.com/WordPress/WordPress/blob/master/wp-admin/theme-editor.php#L124

**Comment by enejb** [comment]
I have tested this on MU and it work.

**Pull request merged**